keras_nlp/samplers/sampler.py

```diff
@@ -143,8 +145,16 @@ def token_probability_fn(inputs, mask):
     def __init__(
         self,
         jit_compile=True,
+        run_eagerly=False,
     ):
+        if run_eagerly and jit_compile:
```

Comment:
Do we need two flags or just one then? What happens if they are both `False`---what is a "non-XLA" graph? I'm confused!

Reply:
This is following the style of `model.compile()`: [code link](https://github.com/keras-team/keras/blob/e6784e4302c7b8cd116b74a784f4b78d60e83c26/keras/engine/training.py#L756)

non-XLA graph is quite common, anything annotated with `tf.function` without `jit_compile=True` is a non-xla graph.

Reply:
Yes, agree, was just trying to understand the difference